pages/index.js

```diff
@@ -57,20 +57,164 @@ const ItemCard = styled.div`
+  margin-bottom: 20px;
+  padding: 10px;
+  border: 1px solid #ddd;
+  border-radius: 5px;
+
+  label {
+    display: block;
+    margin-bottom: 5px;
+  }
+
+  input,
+  select {
+    width: 100%;
+    padding: 8px;
+    margin-bottom: 10px;
+    border-radius: 5px;
+    border: 1px solid #ddd;
+  }
+
+  button {
+    padding: 10px 20px;
+    background-color: #007bff;
+    color: white;
+    border: none;
+    border-radius: 5px;
+    cursor: pointer;
+
+    &:hover {
+      background-color: #0056b3;
+    }
+  }
+
+  .error {
+    color: red;
+    margin-bottom: 10px;
+  }
+`;
+
 const ShoppingList = () => {
-  const [items] = useState(
-    initialItems.map((item) => ({
-      ...item,
-      category: item.category.toLowerCase(),
-    }))
-  );
+  const [items, setItems] = useState([]);
   const [categories] = useState(initialCategories);
+  const [newItem, setNewItem] = useState({
+    name: "",
+    quantity: 1,
+    category: "",
+    comment: "",
+  });
+  const [errors, setErrors] = useState({});
+
+  useEffect(() => {
+    const storedItems = localStorage.getItem("shoppingItems");
+    if (storedItems) {
+      setItems(JSON.parse(storedItems));
+    } else {
+      setItems(
+        initialItems.map((item) => ({
+          ...item,
+          category: item.category.toLowerCase(),
+        }))
+      );
+    }
+  }, []);
+
+  useEffect(() => {
+    if (items.length > 0) {
+      localStorage.setItem("shoppingItems", JSON.stringify(items));
+    }
+  }, [items]);
+
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setNewItem({ ...newItem, [name]: value });
+  };
+
+  const validateForm = () => {
+    const errors = {};
+    if (!newItem.name) errors.name = "Name is required";
+    if (!newItem.quantity || newItem.quantity <= 0)
+      errors.quantity = "Quantity must be a positive number";
+    if (!newItem.category || newItem.category === "default")
+      errors.category = "Please select a category";
+    return errors;
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const formErrors = validateForm();
+    if (Object.keys(formErrors).length > 0) {
+      setErrors(formErrors);
+    } else {
+      setErrors({});
+      const updatedItems = [
+        { ...newItem, id: new Date().getTime().toString() },
+        ...items,
+      ];
+      setItems(updatedItems);
+      setNewItem({ name: "", quantity: 1, category: "", comment: "" });
+    }
+  };
 
   const totalItems = items.reduce((total, item) => total + item.quantity, 0);
 
   return (
     <StyledShoppingList>
       <h1>Shopping List: What I Need to Buy</h1>
+
+      <FormContainer onSubmit={handleSubmit}>
+        <h2>Add New Shopping Item</h2>
+        <div className="error">{errors.name}</div>
+        <label htmlFor="name">Item Name</label>
+        <input
+          type="text"
+          id="name"
+          name="name"
+          value={newItem.name}
+          onChange={handleInputChange}
+        />
```

Comment:
- Instead of manually validating required fields like `name` and `category` in your `validateForm` function, you could leverage the HTML `required` attribute on your inputs. 
- If it is no longer a controlled component, you do not need to specify `value` and `onChange` here.

**This example applies to all other inputs and the select as well.**

This will give users instant feedback and reduce the amount of manual validation you need to implement.
```suggestion
        <input
          type="text"
          id="name"
          name="name"
          required
        />
```